docs/data-types.md

```diff
@@ -32,6 +32,8 @@ Sequelize.DECIMAL(10, 2)              // DECIMAL(10,2)
 Sequelize.DATE                        // DATETIME for mysql / sqlite, TIMESTAMP WITH TIME ZONE for postgres
 Sequelize.DATE(6)                     // DATETIME(6) for mysql 5.6.4+. Fractional seconds support with up to 6 digits of precision
 Sequelize.DATEONLY                    // DATE without time.
+Sequelize.INTERVAL                    // Duration of time. PostgreSQL only.
```

Comment:
Other DBs have a notion of an interval or duration type. E.g. Oracle has `INTERVAL YEAR [(year_precision)] TO MONTH` and `INTERVAL DAY [(day_precision)] TO SECOND [(fractional_seconds)]`.  This should be generalizable to that. 

Note that many DBs, including PG, take a `fields` option that modifies the interval. For example, `INTERVAL YEAR TO MONTH`. I think supporting that should be out of scope.